examples/tests/stack_size_test02/main.c

```diff
@@ -2,7 +2,12 @@ int main(void) {
   printf("Stack Test App\n");
 
   register uint32_t stack_pointer;
+
+#if defined(__thumb__)
   asm volatile ("MOV %0, sp\n" : "=r" (stack_pointer));
+#elif defined(__riscv)
+  asm volatile ("mv %0, sp\n" : "=r" (stack_pointer));
+#endif
```

Comment:
```suggestion
#elif defined(__riscv)
  asm volatile ("mv %0, sp\n" : "=r" (stack_pointer));
#else
#error Unknown architecture
#endif
```

Here and elsewhere, I think once we're doing multi-arch stuff like this we should add the last case to catch anything that might get missed in the future.